pkg/dependency/parser/conda/environment/parse.go

```diff
@@ -0,0 +1,83 @@
+package environment
+
+import (
+	"sort"
+	"strings"
+
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/aquasecurity/trivy/pkg/dependency/types"
+	"github.com/aquasecurity/trivy/pkg/log"
+	xio "github.com/aquasecurity/trivy/pkg/x/io"
+)
+
+type environment struct {
+	Dependencies []Dependency `yaml:"dependencies"`
+}
+
+type Dependency struct {
+	Value string
+	Line  int
+}
+
+type Parser struct {
+	logger *log.Logger
+}
+
+func NewParser() types.Parser {
+	return &Parser{
+		logger: log.WithPrefix("conda"),
+	}
+}
+
+func (p *Parser) Parse(r xio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var env environment
+	if err := yaml.NewDecoder(r).Decode(&env); err != nil {
+		return nil, nil, xerrors.Errorf("unable to decode conda environment.yml file: %w", err)
+	}
+
+	var libs []types.Library
+	for _, dep := range env.Dependencies {
+		lib, err := p.toLibrary(dep)
+		if err != nil {
+			return nil, nil, xerrors.Errorf("unable to parse dependency: %w", err)
+		}
+		libs = append(libs, lib)
+	}
+
+	sort.Sort(types.Libraries(libs))
+	return libs, nil, nil
+}
+
+func (p *Parser) toLibrary(dep Dependency) (types.Library, error) {
+	// Default format for files created using the `conda Export` command: `<Name>=<Version>=<Build>
+	// e.g. `bzip2=1.0.8=h998d150_5`
+	// But it is also possible to set only the dependency name
+	ss := strings.Split(dep.Value, "=")
```

Comment:
It seems like we have to care about a variety of cases.
https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html#package-match-specifications

```
numpy
numpy 1.8*
numpy 1.8.1
numpy >=1.8
numpy ==1.8.1
numpy 1.8|1.8*
numpy >=1.8,<2
numpy >=1.8,<2|1.9
numpy 1.8.1 py27_0
numpy=1.8.1=py27_0
```

Reply:
I updated logic - https://github.com/aquasecurity/trivy/pull/6569/commits/32b7ae23279b53ef769be4855350645006bb1308
@knqyf263 I couldn't create the logic without `regexp`.
But if you have any ideas, please tell me.